### PR TITLE
Blog App: validations, model specs, and n+1 problems

### PR DESCRIPTION
**Pull request by oliverSCZ**
**I did the following:**

- Added validations for the User model and Post model

- Added unit specs for all of your models' methods and validations.

- Added flash messages in the create actions in all your controllers.

### CODE REVIEW

File: app/views/comments/new.html.erb [hunk 1]
```diff
@@ -1,7 +1,10 @@
...
-    <%= form.text_area(:text, size: "30x10") %>
-    <%= form.submit "Submit Comment" %>
+<h2 class="mb-2vh">Write a new comment:</h2>
+<div class="mb-3">
+  <%= form_with(model: @comment, url: user_post_comments_path, method: :post, local: true, class:"row g-3") do |form| %>
+    <%= form.label(:text, "Comment", class:"form-label") %>
+    <%= form.text_area(:text, class:"form-control") %>
+    <div class="col-auto">
+    <%= form.submit "Submit Comment"%>
+    </div>
```
Comment (ahmetbozaci) [inline app/views/comments/new.html.erb]: - [x] I am getting an error when ı try the create comment please fix the error.

- [x] Also, ı am getting this error when clicking on the see all posts, like button. so ı can't check if flash messages is working or not for post#create and likes#create.

image

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -19,6 +19,13 @@ def create
     end
   end
 
+  def destroy
+    @comment = Comment.find(params[:id])
+    @comment.destroy
+
+    redirect_to root_path, notice: 'Comment Deleted!'
+  end
+
```
Comment (ahmetbozaci) [inline app/controllers/comments_controller.rb]: - [x] You should add flash messages in the create actions in all your controllers. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Add%20flash%20messages%20in%20the%20create%20actions%20in%20all%20your%20controllers.)
- [x] Check if the flash message work or not in the view.

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,6 +1,7 @@
 class PostsController < ApplicationController
   def index
-    @user_id = User.find params.require(:user_id)
+    @user_id = User.find(params[:user_id])
+    # @user_id = User.find params.require(:user_id)
```
Comment (ahmetbozaci) [inline app/controllers/posts_controller.rb]: - [x] Please delete the unused comment.
- [x] For solve the n+1 problem you should write `posts = @user.posts.includes(:comments)`   `@posts = @user.posts`

in your case you are using `@user_posts = Post.where(author_id: @user_id)` so you can add `includes(comments)`
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe)

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @oliverSCZ  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.

___ 

### 💯  Nice to have 💯
___


### ♻️ Request changes ♻️

- [x] _Check the comments under the review._
- [x] You should add a flash message to your likes controller.
- [x] Please check if flash messages are working or not in view. [Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md#:~:text=Make%20sure%20that%20the%20N%2B1%20problem%20is%20solved%20when%20fetching%20all%20posts%20and%20their%20comments%20for%20a%20user.%20This%20is%20the%20corresponding%20wireframe)

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- [ ] While your view change request test should also change. That is why ı am getting errors from request tests. Please fix it.

image

___

### Cheers and Happy coding! 💻 👏

almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
## Hi @oliverSCZ, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Required Changes ♻️
- [ ] I was not able to check if you addressed the change requested by the previous reviewer because you changed foreign_key from `user_id` to `author_id` and I am not able to create a database because they are some migrations files you are still using `user_id` please address.
migration



### Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by oliverSCZ** [comment]
The migrations are working properly if you have a fresh database with this name. I've already tried re-installing the project from 0 and it's working. Maybe you already have a DB called 'blog_development'?
But thank you for your feedback...

File: app/models/user.rb [hunk 4]
```diff
@@ -1,12 +1,12 @@
 class User < ApplicationRecord
-  has_many :comments
```
Comment (SajjadAhmad14) [inline app/models/user.rb:2]: Migrations are failing as it's pointed out by the previous reviewer. Kindly fix the issue.
migrate

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Oliver,
### Great work on making the changes requested by a previous reviewer 👏 
### You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### Please remember to tag me in your question so I can receive the notification.
**_Note: Please do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AshakaE — APPROVED** [review]
Hi @oliverSCZ , 

Thanks for making the requested changes 👏 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**